### PR TITLE
Fix/prev next links issue

### PR DESCRIPTION
**Pull request by MaddyDev**
Addressing #10501 
Previous and next navigation buttons are rendered correctly only on one viewlet at any point of time, depending on which one has the last book opened. That's because every time we open a book, we're registering a new navigationProvider with the same providerid which is basically overriding the existing ones. 

In this PR I added a different provider for each viewlet and while retrieving the correct provider used the existing context key to get the appropriate one.

Am not sure if there are better ways to do this, so would like your feedback for improving this.

### CODE REVIEW

File: extensions/notebook/src/book/bookTreeView.ts [hunk 1]
```diff
@@ -259,8 +257,9 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 		});
 	}
 
-	openNotebookAsUntitled(resource: string): void {
+	async openNotebookAsUntitled(resource: string): Promise<void> {
 		try {
+			await vscode.commands.executeCommand('setContext', 'unsavedBooks', true);
```
Comment (corivera) [inline extensions/notebook/src/book/bookTreeView.ts:262]: Should we have some constants for these command strings?

File: extensions/notebook/src/test/book/book.test.ts [hunk 2]
```diff
@@ -42,6 +42,8 @@ export function equalBookItems(book: BookTreeItem, expectedBook: IExpectedBookIt
 	}
 }
 
+const Nav_Provider_Id = 'BookNavigator.Notebooks';
```
Comment (chlafreniere) [inline extensions/notebook/src/test/book/book.test.ts:45]: can we just reference the exported enum here

File: extensions/notebook/src/book/bookTreeView.ts [hunk 3]
```diff
@@ -37,24 +37,27 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
...
 	private _bookTrustManager: IBookTrustManager;
+	providerId: string;
 
 	private _bookViewer: vscode.TreeView<BookTreeItem>;
 	public viewId: string;
 	public books: BookModel[];
 	public currentBook: BookModel;
 
-	constructor(private _apiWrapper: ApiWrapper, workspaceFolders: vscode.WorkspaceFolder[], extensionContext: vscode.ExtensionContext, openAsUntitled: boolean, view: string) {
+	constructor(private _apiWrapper: ApiWrapper, workspaceFolders: vscode.WorkspaceFolder[], extensionContext: vscode.ExtensionContext, openAsUntitled: boolean, view: string, navProviderId: string) {
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/book/bookTreeView.ts:47]: Just make navProviderId a private var here and get rid of providerId

File: extensions/notebook/src/book/bookTreeView.ts [hunk 4]
```diff
@@ -37,24 +37,27 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 	private _initializeDeferred: Deferred<void> = new Deferred<void>();
 	private _openAsUntitled: boolean;
 	private _bookTrustManager: IBookTrustManager;
+	providerId: string;
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/book/bookTreeView.ts:40]: Not needed with the change suggested below - but FYI please always specify visibility modifiers when defining instance level vars.

File: src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts [hunk 5]
```diff
@@ -220,7 +222,15 @@ export class NotebookService extends Disposable implements INotebookService {
 	}
 
 	getNavigationProvider(): INavigationProvider {
-		let provider = this._navigationProviders.size > 0 ? this._navigationProviders.values().next().value : undefined;
+		let provider;
+		if (this._navigationProviders.size > 0) {
+			let unsavedBooks = this.contextKeyService.getContextKeyValue('unsavedBooks');
+			let providerName = NavigationProviders.NotebooksNavigator;
```
Comment (Charles-Gagnon) [inline src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts:228]: `const providerName = unsavedBooks ? NavigationProviders.ProvidedBooksNavigator : NavigationProviders.NotebooksNavigator;`

File: src/sql/workbench/services/notebook/browser/notebookService.ts [hunk 6]
```diff
@@ -34,6 +34,13 @@ export interface ILanguageMagic {
 	executionTarget?: string;
 }
 
+/**
+		 * The valid navigation providers.
```
Comment (Charles-Gagnon) [inline src/sql/workbench/services/notebook/browser/notebookService.ts:38]: alignment

File: extensions/notebook/package.json [hunk 7]
```diff
@@ -186,7 +186,7 @@
         }
       },
       {
-        "command": "notebook.command.searchUntitledBook",
+        "command": "notebook.command.searchProvidedBook",
```
Comment (chlafreniere) [inline extensions/notebook/package.json:189]: are there any places where we're referencing notebook.command.searchUntitledBook?

File: extensions/notebook/src/book/bookTreeView.ts [hunk 8]
```diff
@@ -493,4 +496,30 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 			default: false
 		});
 	}
+
+	getNavigation(uri: vscode.Uri): Thenable<azdata.nb.NavigationResult> {
+		let result: azdata.nb.NavigationResult;
+		//let book = this.getBookFromItemPath(uri.scheme === 'untitled' ? vscode.Uri.file(uri.path).path : uri.path);
```
Comment (chlafreniere) [inline extensions/notebook/src/book/bookTreeView.ts:502]: if this is commented out it should just be removed.

File: extensions/notebook/src/extension.ts [hunk 9]
```diff
@@ -120,9 +118,9 @@ export async function activate(extensionContext: vscode.ExtensionContext): Promi
 	}
 
 	let workspaceFolders = vscode.workspace.workspaceFolders?.slice() ?? [];
-	const bookTreeViewProvider = new BookTreeViewProvider(appContext.apiWrapper, workspaceFolders, extensionContext, false, BOOKS_VIEWID);
+	const bookTreeViewProvider = new BookTreeViewProvider(appContext.apiWrapper, workspaceFolders, extensionContext, false, BOOKS_VIEWID, NavigationProviders.NotebooksNavigator);
```
Comment (chlafreniere) [inline extensions/notebook/src/extension.ts:121]: this is totally random but this reminds me of a baseball team near where i grew up:

[Norwich naviagators](http://logotypes101.com/logos/435/BCAB94B69EE1DD3FC57A0E1113AB50CF/Norwich_Navigators86.png)

File: extensions/notebook/package.json (hunk 7 above)
Comment (MaddyDev) [reply extensions/notebook/package.json:189]: we use it for search on the items in Provided Books viewlet.

Comment (chlafreniere) [reply extensions/notebook/package.json:189]: ok. just wanted to make sure that anything that referenced the old command name has been updated.

**Pull request merged**